### PR TITLE
UCT/TCP: Add dummy memory registration

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Add dummy memory registration

## Why ?

To make possible select TCP for RMA_BW lanes when it will support PUT ZCOPY operation

## How ?

Apply the same approach as SELF did

### CODE REVIEW

File: src/uct/sm/self/self.c [hunk 1]
```diff
@@ -347,12 +347,12 @@ static ucs_status_t uct_self_md_open(uct_component_t *component, const char *md_
...
 {
     static uct_md_ops_t md_ops = {
-        .close              = (void*)ucs_empty_function,
+        .close              = ucs_empty_function,
         .query              = uct_self_md_query,
         .mkey_pack          = ucs_empty_function_return_success,
         .mem_reg            = uct_self_mem_reg,
         .mem_dereg          = ucs_empty_function_return_success,
-        .detect_memory_type = (void *)ucs_empty_function_return_zero,
+        .detect_memory_type = ucs_empty_function_return_success
```
Comment (yosefe) [inline src/uct/sm/self/self.c]: should be ucs_empty_function_return_unsupported

File: src/uct/tcp/tcp_md.c [hunk 2]
```diff
@@ -27,17 +30,25 @@ static ucs_status_t uct_tcp_md_query(uct_md_h md, uct_md_attr_t *attr)
...
         .close              = ucs_empty_function,
         .query              = uct_tcp_md_query,
-        .mkey_pack          = ucs_empty_function_return_unsupported,
-        .mem_reg            = ucs_empty_function_return_unsupported,
-        .mem_dereg          = ucs_empty_function_return_unsupported,
-        .detect_memory_type = ucs_empty_function_return_unsupported,
+        .mkey_pack          = ucs_empty_function_return_success,
+        .mem_reg            = uct_tcp_md_mem_reg,
+        .mem_dereg          = ucs_empty_function_return_success,
+        .detect_memory_type = ucs_empty_function_return_success
```
Comment (yosefe) [inline src/uct/tcp/tcp_md.c]: ucs_empty_function_return_unsupported

File: src/uct/sm/self/self.c (hunk 1 above)
Comment (dmitrygx) [reply src/uct/sm/self/self.c]: done

File: src/uct/tcp/tcp_md.c (hunk 2 above)
Comment (dmitrygx) [reply src/uct/tcp/tcp_md.c]: done

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12083/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>UNKNOWN</b> on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 04-Sep-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/5a6836fc95a18d9f795161e3f18036fa704a989f/_main) | :question: ABORTED
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/09049e700f9f916bb175448df46b1ac01929ec30/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/ce1f2842d33f22ac985fe0b9fc868c45fdd6357e/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/86a685a951cac5c59a160385d0f6b9d1350d5866/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/d0010b5927da2cbe1a5942e06231846e008f891b/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/7938d4f40d08fc5301fc834107ed14caf31f8b93/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/b558a6d082b007e40627fe7ef012db9df28a6a4b/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/4590357b5980590d8ebeb7c27e5100f7cb5ec1de/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/79e93f8e8abc930d13be6c99d087a6a6d0057f63/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/31ded224db92502ab97803d7bc02bef7ed8428e8/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/142d1ce0f949c24d86550dcc2d894f86eed80a02/hpc-arm-cavium-jenkins_W0) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/0e34dce6fc803db8e12a0c4b984224239138e94f/hpc-arm-cavium-jenkins_W1) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/a62a79c48e3a281f98b989bdd715b287466802ac/hpc-arm-cavium-jenkins_W2) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/85d220b51243e964e215e044602eb735db53055d/hpc-arm-cavium-jenkins_W3) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/a5e0a70ff94e7a4a3c53634e56ba5d6112dfe49a/hpc-arm-hwi-jenkins_W0) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/a8d2c3169970969b5ee58c60bdb78fd125b86e22/hpc-arm-hwi-jenkins_W1) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/1c5ad609fd0b890bfc7321242f60f83e9c75d112/hpc-arm-hwi-jenkins_W2) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/872106e87c3336f3c83f89929b951b747041ecdc/hpc-arm-hwi-jenkins_W3) | :question: UNKNOWN
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/0bb3a3d7b5da08386067107b6dbe700ce7581603/hpc-test-node-gpu_W1) | :question: UNKNOWN
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/86b28617c6e2cd9969215af48e07c9786328212f/hpc-test-node-gpu_W2) | :question: UNKNOWN
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/d3408c1d59a925c8258f925debe30cf324517f5a/hpc-test-node-gpu_W3) | :question: UNKNOWN
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/7bc114f91b8534be0b51491b84d7932d16d26cd5/hpc-test-node-legacy_W0) | :question: UNKNOWN
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/4dc1358510a232972bf4bdcb893fe04c85935543/hpc-test-node-legacy_W1) | :question: UNKNOWN
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/2d5f51032b7daf7ff246ce447f9b735357774316/hpc-test-node-legacy_W2) | :question: UNKNOWN
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/c710cc484aef5dfc4cdc1f2077feb599b59bfcb4/hpc-test-node-legacy_W3) | :question: UNKNOWN
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/f5d9fef27d59a51852386952dfde9a871f33fa51/hpc-test-node-new_W0) | :question: UNKNOWN
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/0c034dc1ef45e4679a718c54c1ae272b17389ab9/hpc-test-node-new_W1) | :question: UNKNOWN
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/87e12db7a51137b3463a78851249cb897e03940f/hpc-test-node-new_W2) | :question: UNKNOWN
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/1812f999dde2f04737599d43cbe81eab/raw/33eead035818659858c962c5e7cc570b5e9f88a9/hpc-test-node-new_W3) | :question: UNKNOWN

**Comment by dmitrygx** [comment]
bot:pipe:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12085/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 04-Sep-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/e72864333baaf442a414c82a6bb3ec6e3b773c37/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/e32406510dd650b1b090626b780d1b72020d361d/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/31f3b4ae56878fd82e7449106cba64dd02add147/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/68785e91da1a00e9b2bfab2d42de38b809bbd3e6/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/459b960c001ed0a3720028b90c5d72dba3f5b817/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/a6ba148f390d2a86479e09dded2da027c8edf250/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/fc0bb6503575dae469258073dd175f87b1fbb777/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/bec0b0c448fc8473940399cb858a277b812fd469/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/24052c093c5c4308c8a810ff3caa1b13bc821cc8/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/645e9dafcb614512fe6839b615bb3947cf575639/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/43dfe3e61f161388a895a24edd7013e6bf79c296/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/44ccd59cfca2e18f41b216b922d2106a0bda25a1/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/d758fef6823557e6a79e1bc65b12557d56c8fce3/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/8d2a48812f8a3895503a9c004480ee6929296a4f/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/42f06cb0f8ac0b713a9b04c407f5ee97c637426d/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/43a98ca56f87e238d116f47d86e453eeeca47b1d/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/e046c6737d5915d5372027ec314006d8309b0461/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/58d671325dd256ca20b4491faf90187a7c7ce9af/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/53da38f9c43822a04da973c49206639e391159fe/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/46dc5cefbcf4e1be7e03d7448b1d93678f58e8a7/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/d7519dcc7270b170b00ac7cef562b91470f5d5a8/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/953a056db1acc0803b43179e488dacb6619b1b3a/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/5a1baa299f95c996817295c5ca21577b9b8d08af/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/e235c36ce066f26a7e4b900bb4c9f7140948272b/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/1af2395df0ba23814c6548e3b36a542d99d8995b/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/77b426cdea30e3f6156604898723fcffa3f99cbe/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/886257a6efb9797de0bc501ba5175facc006a5b0/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/f26b8d09d9c28260addbbcd76d8153b3bdbf2773/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/5c2c02c0f0c526d3b600d1d5e9ef3339/raw/a5a30106945337b179b70565c1ab549c43fcb077/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by dmitrygx** [comment]
@yosefe ok to merge?

**Pull request merged**